### PR TITLE
Added OWTF project ideas for GSoC 2021

### PR DESCRIPTION
**Pull request by sharmamohit123**

### CODE REVIEW

File: pages/initiatives/gsoc/gsoc2021ideas.md [hunk 1]
```diff
@@ -235,3 +235,136 @@ OWASP IIDS is an open source software that leverages the benefits of Artificial
...
+##### LEGAL CLARIFICATION (Just in case!): 
+The passive online scanner, simply makes OWTF passive testing through third party websites more accessible to anybody, however it is the user that must 
+1. click the link manually + 
+2. do something bad with that afterwards + 
+3. doing 1 + 2 WITHOUT permission :). 
+Therefore this passive online scanner does not do anything illegal 
+[More information about why this is not illegal here](http://www.slideshare.net/abrahamaranguren/legal-and-efficient-web-app-testing-without-permission) 
+(recommended reading!)
+
+For background on OWASP OWTF please see: [OWASP OWTF](https://www.owasp.org/index.php/OWASP_OWTF)
```
Comment (kingthorin) [inline pages/initiatives/gsoc/gsoc2021ideas.md:265]: Should update these... throughout the OWTF additions

Comment (kingthorin) [reply pages/initiatives/gsoc/gsoc2021ideas.md:265]: Looks like OWASP.org content needs an update it still talks about gsoc 2020 and release 2.3b

**Review by kingthorin — CHANGES_REQUESTED** [review]

File: pages/initiatives/gsoc/gsoc2021ideas.md (hunk 1 above)
Comment (sharmamohit123) [reply pages/initiatives/gsoc/gsoc2021ideas.md:265]: Added a PR https://github.com/OWASP/www-project-owtf/pulls. Pls have a look!

Comment (kingthorin) [reply pages/initiatives/gsoc/gsoc2021ideas.md:265]: Tks, the links here should still be updated

Comment (kingthorin) [inline pages/initiatives/gsoc/gsoc2021ideas.md:265]:
```suggestion
For background on OWASP OWTF please see: [OWASP OWTF](https://owasp.org/www-project-owtf/)
```

File: pages/initiatives/gsoc/gsoc2021ideas.md [hunk 2]
```diff
@@ -235,3 +235,136 @@ OWASP IIDS is an open source software that leverages the benefits of Artificial
...
+* [Mohit Sharma](mailto:ms892075@gmail.com)
+
+
+#### OWASP OWTF - Web interface enhancements
+
+##### Brief Explanation:
+
+The current owtf web interface is implemented in ReactJs with Redux as the state manager. The project involves - (1) integration of Typescript in the code to ease the refactoring process, (2) upgrading the UI to remove additional dependencies and improve user experience . Check out the current implementation of the web interface at [OWTF on GitHub](https://github.com/owtf/owtf/tree/develop).
+
+For background on OWASP OWTF please see: [OWASP_OWTF](https://www.owasp.org/index.php/OWASP_OWTF)
```
Comment (kingthorin) [inline pages/initiatives/gsoc/gsoc2021ideas.md:291]:
```suggestion
For background on OWASP OWTF please see: [OWASP_OWTF](https://owasp.org/www-project-owtf/)
```

File: pages/initiatives/gsoc/gsoc2021ideas.md [hunk 3]
```diff
@@ -235,3 +235,136 @@ OWASP IIDS is an open source software that leverages the benefits of Artificial
...
+* [Mohit Sharma](mailto:ms892075@gmail.com)
+
+
+#### OWASP OWTF - Login/Signup Implementation
+
+##### Brief Explanation:
+
+Some pages of the new OWTF interface has been under progess for a very long time.  Complete implementation of the Login/Signup Page (APIs + frontend) with proper unit/functional tests will be deliverable for this project. Check out the current implementation of the web interface at [OWTF on GitHub](https://github.com/owtf/owtf/tree/develop).
+
+For background on OWASP OWTF please see: [OWASP_OWTF](https://www.owasp.org/index.php/OWASP_OWTF)
```
Comment (kingthorin) [inline pages/initiatives/gsoc/gsoc2021ideas.md:320]:
```suggestion
For background on OWASP OWTF please see: [OWASP_OWTF](https://owasp.org/www-project-owtf/)
```

File: pages/initiatives/gsoc/gsoc2021ideas.md [hunk 4]
```diff
@@ -235,3 +235,136 @@ OWASP IIDS is an open source software that leverages the benefits of Artificial
...
+
+
+#### OWASP OWTF - General Improvements
+
+##### Brief Explanation:
+
+There are many small but important enhancements in the issue tracker which are too small to make a single project, but they can be grouped together to make a suitable GSoC project.
+The aim of the project is to implement some of the enhancements suggested in the issue tracker to improve user experience (adding new useful features and making the owtf tool easier to use), security and performance.
+
+For background on OWASP OWTF please see: [OWASP_OWTF](https://www.owasp.org/index.php/OWASP_OWTF)
```
Comment (kingthorin) [inline pages/initiatives/gsoc/gsoc2021ideas.md:350]:
```suggestion
For background on OWASP OWTF please see: [OWASP_OWTF](https://owasp.org/www-project-owtf/)
```

**Review by kingthorin — APPROVED** [review]

**Pull request merged**